src/protocols/mcsp/MessageCounterManager.cpp

```diff
@@ -60,7 +60,72 @@ void MessageCounterSyncMgr::Shutdown()
     }
 }
 
-void MessageCounterSyncMgr::OnMessageReceived(Messaging::ExchangeContext * exchangeContext, const PacketHeader & packetHeader,
+bool MessageCounterManager::IsSyncCompleted(Transport::PeerMessageCounter & counter)
+{
+    return counter.IsSyncCompleted();
+}
+
+CHIP_ERROR MessageCounterManager::VerifyCounter(Transport::PeerMessageCounter & counter, const PacketHeader & packetHeader)
+{
+    return counter.Verify(packetHeader.GetMessageId());
+}
+
+void MessageCounterManager::CommitCounter(Transport::PeerMessageCounter & counter, const PacketHeader & packetHeader)
+{
+    counter.Commit(packetHeader.GetMessageId());
+}
+
+CHIP_ERROR MessageCounterManager::StartSync(SecureSessionHandle session, Transport::PeerConnectionState * state)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // Initiate message counter synchronization if no message counter synchronization is in progress.
+    if (!state->GetSessionMessageCounter().GetPeerMessageCounter().IsSyncStarted())
+    {
+        err = SendMsgCounterSyncReq(session, state);
+    }
+
+    return err;
+}
+
+CHIP_ERROR MessageCounterManager::QueueSendMessageAndStartSync(SecureSessionHandle session, Transport::PeerConnectionState * state,
+                                                               PayloadHeader & payloadHeader, System::PacketBufferHandle msgBuf)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // Queue the message as needed for sync with destination node.
+    err = AddToRetransmissionTable(session, state->GetPeerNodeId(), payloadHeader, std::move(msgBuf));
```

Comment:
nit: you can do this to have shorter code (and one less variable):

```cpp
ReturnErrorOnFailure(
    AddToRetransmissionTable(session, state->GetPeerNodeId(), payloadHeader, std::move(msgBuf)));
    
ReturnErrorOnFailure(StartSync(session, state));

return CHIP_NO_ERROR;
```

same below. The intent of 'return' macros was to avoid having extra CHIP_ERROR assignments and variables.

Reply:
done